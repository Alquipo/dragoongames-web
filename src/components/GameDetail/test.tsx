// import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import GameDetails from '.'

describe('<GameDetails />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<GameDetails />)

    // expect(screen.getByRole('heading', { name: /GameDetails/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
