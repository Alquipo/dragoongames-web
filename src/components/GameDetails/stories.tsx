import { Story, Meta } from '@storybook/react/types-6-0'
import GameDetail, { GameDetailsProps } from '.'
import mockGame from './mock'

export default {
  title: 'Game/GameDetail',
  component: GameDetail,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  args: mockGame,
  argTypes: {
    releaseDate: {
      control: 'date'
    },
    platforms: {
      control: {
        type: 'inline-check',
        options: ['windows', 'linux', 'mac']
      }
    },
    genres: {
      control: {
        type: 'inline-check',
        options: ['Role-playing', 'Narrative', 'Action']
      }
    }
  }
} as Meta

export const Default: Story<GameDetailsProps> = (args) => (
  <div style={{ maxWidth: '130rem', margin: '0 auto' }}>
    <GameDetail {...args} />
  </div>
)
