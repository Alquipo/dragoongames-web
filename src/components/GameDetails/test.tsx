import { render, screen } from 'utils/test-utils'

import GameDetails, { GameDetailsProps } from '.'

const props: GameDetailsProps = {
  developer: 'Different Tales',
  publisher: 'anything',
  platforms: ['linux', 'mac', 'windows'],
  releaseDate: '2020-11-21T23:00:00',
  rating: 'BR0',
  genres: ['Role-playing', 'Narrative']
}

describe('<GameDetails />', () => {
  it('should render the blocks', () => {
    render(<GameDetails {...props} />)

    expect(
      screen.getByRole('heading', { name: /Developer/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /release date/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Platform/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Publisher/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /Rating/i })).toBeInTheDocument()

    expect(screen.getByRole('heading', { name: /Genres/i })).toBeInTheDocument()
  })

  it('should render platform icons', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByRole('img', { name: /linux/i })).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /windows/i })).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /mac/i })).toBeInTheDocument()
  })

  it('should render free rating when BR0', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByText(/free/i)).toBeInTheDocument()
  })

  it('should render the Developer', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByText(/Different Tales/i)).toBeInTheDocument()
  })

  it('should render the publisher', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByText(/anything/i)).toBeInTheDocument()
  })

  it('should render 18+ free rating when BR18', () => {
    render(<GameDetails {...props} rating="BR18" />)

    expect(screen.getByText(/18\+/i)).toBeInTheDocument()
  })

  it('should render the formatted date', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByText('Nov 21, 2020')).toBeInTheDocument()
  })

  it('should render a list of genres', () => {
    render(<GameDetails {...props} />)

    expect(screen.getByText('Role-playing / Narrative')).toBeInTheDocument()
  })
})
