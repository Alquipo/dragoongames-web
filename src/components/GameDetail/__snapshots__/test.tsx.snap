// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameDetails /> should render the heading 1`] = `
.c2 {
  color: #FAFAFA;
  padding-left: 0.8rem;
  border-left: 0.7rem solid #3CD3C1;
  font-size: 2.0rem;
}

.c1 {
  display: none;
}

.c0 {
  margin: 2.4rem 0;
}

.c3 {
  display: grid;
  gap: 1.6rem;
  grid-template-columns: repeat(2,1fr);
  margin-top: 2.4rem;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
}

.c5 {
  font-size: 1.6rem;
  font-weight: 600;
  color: #FAFAFA;
}

@media (min-width:768px) {
  .c2 {
    font-size: 2.8rem;
  }
}

@media (min-width:450px) {
  .c1 {
    display: block;
  }
}

@media (min-width:768px) {
  .c3 {
    grid-template-columns: repeat(3,1fr);
  }
}

@media (min-width:1170px) {
  .c3 {
    grid-template-columns: repeat(6,1fr);
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
      color="white"
    >
      Game Details
    </h2>
  </div>
  <div
    class="c3"
  >
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Developer
      </h3>
      <p
        class="c5"
      >
        Gearbox Software
      </p>
    </div>
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Release Date
      </h3>
      <p
        class="c5"
      >
        Nov 16, 2019
      </p>
    </div>
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Platforms
      </h3>
    </div>
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Publisher
      </h3>
      <p
        class="c5"
      >
        2K
      </p>
    </div>
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Rating
      </h3>
      <p
        class="c5"
      >
        18+
      </p>
    </div>
    <div
      class=""
    >
      <h3
        class="c4"
      >
        Genres
      </h3>
      <p
        class="c5"
      >
        Action / Adventure
      </p>
    </div>
  </div>
</div>
`
